File: Domain/Category.cs
using Domain.Enums;

namespace Domain
{
    public class Category
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int ParentCategoryId { get; set; }
        public CategoryStatus Status { get; set; }

        public Category ParentCategory { get; set; }
        public ICollection<Category> ChildCategories { get; set; }
        public ICollection<Product> Products { get; set; }
    }
}

File: client-zespolowy/src/app/models/onlineshop/Category.ts
import { CategoryStatus } from "../enums/CategoryStatus";

export interface Category {
    id: number;
    name: string;
    parentCategoryId?: number;
    status: CategoryStatus;
}

export interface CategoryTree {
    id: number;
    name: string;
    parentCategoryId?: number;
    status: CategoryStatus;

    childCategories: CategoryTree[];
}